PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/mysql/AddMySQLDependencyAction.java

```diff
@@ -0,0 +1,175 @@
+                    final File pomFile = new File(mavenProject.getFile().getCanonicalPath());
+                    ProgressManager.checkCanceled();
+                    if (applyVersionChanges(dependencyManager, pomFile, managerDependencyVersionsMaps, versionChanges)) {
+                        noticeUserVersionChanges(project, pomFile, versionChanges);
+                    } else {
+                        PluginUtil.showInfoNotificationProject(project, "Your project is update-to-date.", "No updates are needed.");
+                    }
+                } catch (DocumentException | IOException | AzureExecutionException | MavenProcessCanceledException e) {
+                    PluginUtil.showErrorNotification("Error",
+                                                     "Failed to update Azure Spring Cloud dependencies due to error: "
+                                                             + e.getMessage());
+                }
+            });
+        return false;
+    }
+
+    public void update(@NotNull AnActionEvent event) {
+        final Presentation presentation = event.getPresentation();
+        final Module module = event.getData(LangDataKeys.MODULE);
+        boolean isMaven = module != null && MavenUtils.isMavenProject(module.getProject());
+        if (isMaven && StringUtils.equals(event.getPlace(), "EditorPopup")) {
+            presentation.setEnabledAndVisible(isEditingMavenPomXml(module, event));
+        } else {
+            presentation.setEnabledAndVisible(isMaven && ModuleTypeId.JAVA_MODULE.equals(module.getModuleTypeName()));
+        }
+    }
+
+    private static List<DependencyArtifact> calculateVersionChanges(Map<String, DependencyArtifact> versionMaps)
+            throws AzureExecutionException, DocumentException, IOException {
+        DependencyArtifact dependency = getDependencyArtifact(GROUP_ID, ARTIFACT_ID, versionMaps);
+
+        if (StringUtils.isBlank(dependency.getCurrentVersion())) {
+            List<String> latestVersions = MavenDependencyUtils.getMavenCentralVersions(GROUP_ID, ARTIFACT_ID);
+            if (latestVersions.isEmpty()) {
+                PluginUtil.showErrorNotification("Error",
+                                                 String.format("Artifact '%s:%s' cannot be found from maven central.", GROUP_ID, ARTIFACT_ID));
+            }
+            dependency.setCompatibleVersion(latestVersions.get(latestVersions.size() - 1));
+
+        }
+        return StringUtils.equals(dependency.getCurrentVersion(), dependency.getCompatibleVersion()) ? Collections.emptyList() :
+               Arrays.asList(dependency);
+    }
+
+    @Override
+    protected String getServiceName(AnActionEvent event) {
+        return TelemetryConstants.MYSQL;
+    }
+
+    @Override
+    protected String getOperationName(AnActionEvent event) {
+        return TelemetryConstants.ADD_MY_SQL_DEPENDENCY;
+    }
+
+    private static boolean applyVersionChanges(SpringCloudDependencyManager dependencyManager,
```

Comment:
could we extract this method to `AzureDependencyActionBase`? Seems the logic could be used for other dependency action

Reply:
Done